servercom/servercom.go

```diff
@@ -0,0 +1,157 @@
+package servercom
+
+import (
+	"fmt"
+	mm "github.com/mattermost/platform/model"
+	"io"
+)
+
+type Client interface {
+    Login(loginId string, password string) (*mm.Result, *mm.AppError)
+    GetTeamByName(teamName string) (*mm.Result, *mm.AppError)
+    SetTeamId(teamId string)
+    GetAllTeamListings() (*mm.Result, *mm.AppError)
+    GetChannelByName(channelName string) (*mm.Result, *mm.AppError)
+    GetChannels(etag string) (*mm.Result, *mm.AppError)
+    CreatePost(post *mm.Post) (*mm.Result, *mm.AppError)
+    GetPostsSince(channelId string, time int64) (*mm.Result, *mm.AppError)
+}
+
+// test that *mm.Client satisfies the Client interface at compile time
+var _ Client = &mm.Client{}
+
+/*
+ServerCom acts as a mitigator between the frontend and the mattermost model API.
+*/
+type ServerCom struct {
+	Output  io.Writer // This is where the print methods will send output
+	Client  Client
+	Team    mm.Team
+	Channel *mm.Channel
+}
+
+/*
+Startup accepts the url and login credentials for a user, and returns a new serverCom struct.
+*/
+func Startup(url string, username string, password string, output io.Writer) (*ServerCom, error) {
```

Comment:
No, this is  essentially the constructor for a `*ServerCom`. The convention is to use camelCase for private functions and PascalCase for public ones. It's not really a convention though, because it's enforced by the compiler. This isn't something that `go fmt` takes care of because it radically affects the code's logic.